src/box/lua/load_cfg.lua

```diff
@@ -257,6 +268,17 @@ local compat_options = {
         obsolete = nil,
         default = 'new',
     },
+    {
+        name = 'wal_cleanup_delay',
+        brief = wal_cleanup_delay_brief,
+        oldval = 4 * 3600,
+        newval = 0,
```

Comment:
What happens if the compat option is new, but the database schema is old? For example, in case of Tarantool 4.0.0 that is started from a snapshot from Tarantool 3.0.0. Is it possible that a snapshot/an xlog that is needed for a replica will be removed on startup of the master?

Reply:
Thanks for pointing the problem out!
Now I automatically set `wal_cleanup_delay` to 4 hours (with warning) if there is no space `_gc_consumers`, `wal_cleanup_delay` is 0 and the instance participates in a cluster.